jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1559,15 +1558,51 @@ ecma_builtin_typedarray_prototype_find (ecma_value_t this_arg, /**< this argumen
 
     if (call_result)
     {
-      return element_value;
+      return is_find ? element_value : ecma_make_uint32_value (buffer_index);
     }
-
+    buffer_index++;
     ecma_free_value (element_value);
   }
 
   return ECMA_VALUE_UNDEFINED;
```

Comment:
`22.1.3.9. 9` Return -1. FindIndex should return -1 if no matching element is found.